static/js/composebox_typeahead.js

```diff
@@ -303,7 +303,7 @@ exports.compose_content_begins_typeahead = function (query) {
         // as :P or :-p
         // Also, if the user has only typed a colon and nothing after,
         // no need to match yet.
-        if (/^:-?.?$/.test(current_token)) {
+        if (/^:-.?$/.test(current_token) || /^:[^a-z+]?$/.test(current_token)) {
```

Comment:
@shubhamdhama can you explain why the `.` and `?` are swapped in this first part of the expression?

Reply:
@timabbott actually first part is covering up(or preventing the emoji typeahead of) the cases like `:-` for `:-1:` like emojis and `:-p` for `:non-potable_water:` like emojis.
Previous regex i.e. `/^:-?.?$/`  was making both `-` as well as `.`(any letter or character) optional for matching but now in `/^:-.?$/` removing `?` before `-` makes only `.`(any letter or character) optional and rest of cases are covered by the second expression.

Reply:
Ahh, OK, makes sense.